BlobStorage/Advanced.cs

```diff
@@ -1523,43 +1269,46 @@ private static async Task CreateBlockBlobSnapshotAsync(CloudBlobContainer contai
         /// <summary>
         /// Gets a reference to a blob created previously, and copies it to a new blob in the same container.
         /// </summary>
-        /// <param name="container">A CloudBlobContainer object.</param>
+        /// <param name="container">A BlobContainerClient object.</param>
         /// <returns>A Task object.</returns>
-        private static async Task CopyBlockBlobAsync(CloudBlobContainer container)
+        private static async Task CopyBlockBlobAsync(BlobContainerClient container)
         {
-            CloudBlockBlob sourceBlob = null;
-            CloudBlockBlob destBlob = null;
+            BlockBlobClient sourceBlob = null;
+            BlockBlobClient destBlob = null;
```

Comment:
I think `BlobBaseClient` is all you need for the operations you're calling. In general, we shouldn't be demonstrating with `BlockBlobClient`, `PageBlobClient`, or `AppendBlobClient` unless we very explicitly need methods on those clients.